velox/expression/tests/CastExprTest.cpp

```diff
@@ -320,6 +320,35 @@ TEST_F(CastExprTest, timestampAdjustToTimezoneInvalid) {
   EXPECT_THROW(testFunc(), std::runtime_error);
 }
 
+TEST_F(CastExprTest, date) {
+  testCast<std::string, Date>(
+      "date",
+      {
+          "1970-01-01",
```

Comment:
nit: perhaps, add some dates in the deep past and future? For example, "1812-04-15" and "2135-11-09".